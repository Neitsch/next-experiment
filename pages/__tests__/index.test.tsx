import { mount, shallow } from "enzyme";
import React, { Children } from "react";
/* tslint:disable-next-line */
import { MockedProvider } from "react-apollo/test-utils";

const ContainerMock = mockData =>
  class MyMock extends React.Component {
    public render() {
      return this.props.children(mockData);
    }
  };

describe("Index page", () => {
  it("Initial props", async () => {
    jest.resetAllMocks();
    jest.resetModules();
    const getLocalCookie = jest.fn().mockImplementation(() => "local");
    const getServerCookie = jest.fn().mockImplementation(() => "server");
    jest.doMock("../../lib/auth/index", () => ({
      getLocalCookie,
      getServerCookie,
    }));
    const Index = require("../index").default;
    process.browser = false;
    const retServer = await Index.getInitialProps({ req: jest.fn() });
    expect(retServer).toEqual({ isAuthenticated: "server" });
    expect(getServerCookie).toHaveBeenCalledTimes(1);
    process.browser = true;
    const retLocal = await Index.getInitialProps();
    expect(retLocal).toEqual({ isAuthenticated: "local" });
    expect(getLocalCookie).toHaveBeenCalledTimes(1);
  });
  it("renders unauth", () => {
    jest.resetAllMocks();
    jest.resetModules();
    jest.doMock("graphql-tag");
    jest.doMock("next/link");
    jest.doMock("react-apollo");
    const Index = require("../index").default;
    expect(shallow(<Index isAuthenticated={false} />)).toMatchSnapshot();
  });
  describe("renders auth", () => {
    it("Loading", () => {
      jest.resetAllMocks();
      jest.resetModules();
      jest.doMock("graphql-tag");
      jest.doMock("next/link");
      jest.doMock("react-apollo", () => ({
        Query: ContainerMock({ data: {}, error: null, loading: true }),
      }));
      const Index = require("../index").default;
      expect(mount(<Index isAuthenticated={true} />)).toMatchSnapshot();
    });
    it("Error", () => {
      jest.resetAllMocks();
      jest.resetModules();
      jest.doMock("graphql-tag");
      jest.doMock("next/link");
      jest.doMock("react-apollo", () => ({
        Query: ContainerMock({ data: {}, error: {}, loading: false }),
      }));
      const Index = require("../index").default;
      expect(mount(<Index isAuthenticated={true} />)).toMatchSnapshot();
    });
    it("Data", () => {
      jest.resetAllMocks();
      jest.resetModules();
      jest.doMock("graphql-tag");
      jest.doMock("next/link");
      jest.doMock("react-apollo", () => ({
        Query: ContainerMock({
          data: { user: "Hi!" },
          error: null,
          loading: false,
        }),
      }));
      const Index = require("../index").default;
      expect(mount(<Index isAuthenticated={true} />)).toMatchSnapshot();
    });
  });
});
