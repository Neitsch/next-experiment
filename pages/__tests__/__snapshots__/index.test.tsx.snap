// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page renders auth Data 1`] = `
<Index
  isAuthenticated={true}
>
  <MyMock>
    <Home
      data={
        Object {
          "user": "Hi!",
        }
      }
    >
      <WithStyles(Card)>
        <Card
          classes={
            Object {
              "root": "MuiCard-root-1",
            }
          }
          raised={false}
        >
          <WithStyles(Paper)
            className="MuiCard-root-1"
            elevation={1}
          >
            <Paper
              className="MuiCard-root-1"
              classes={
                Object {
                  "elevation0": "MuiPaper-elevation0-4",
                  "elevation1": "MuiPaper-elevation1-5",
                  "elevation10": "MuiPaper-elevation10-14",
                  "elevation11": "MuiPaper-elevation11-15",
                  "elevation12": "MuiPaper-elevation12-16",
                  "elevation13": "MuiPaper-elevation13-17",
                  "elevation14": "MuiPaper-elevation14-18",
                  "elevation15": "MuiPaper-elevation15-19",
                  "elevation16": "MuiPaper-elevation16-20",
                  "elevation17": "MuiPaper-elevation17-21",
                  "elevation18": "MuiPaper-elevation18-22",
                  "elevation19": "MuiPaper-elevation19-23",
                  "elevation2": "MuiPaper-elevation2-6",
                  "elevation20": "MuiPaper-elevation20-24",
                  "elevation21": "MuiPaper-elevation21-25",
                  "elevation22": "MuiPaper-elevation22-26",
                  "elevation23": "MuiPaper-elevation23-27",
                  "elevation24": "MuiPaper-elevation24-28",
                  "elevation3": "MuiPaper-elevation3-7",
                  "elevation4": "MuiPaper-elevation4-8",
                  "elevation5": "MuiPaper-elevation5-9",
                  "elevation6": "MuiPaper-elevation6-10",
                  "elevation7": "MuiPaper-elevation7-11",
                  "elevation8": "MuiPaper-elevation8-12",
                  "elevation9": "MuiPaper-elevation9-13",
                  "root": "MuiPaper-root-2",
                  "rounded": "MuiPaper-rounded-3",
                }
              }
              component="div"
              elevation={1}
              square={false}
            >
              <div
                className="MuiPaper-root-2 MuiPaper-elevation1-5 MuiPaper-rounded-3 MuiCard-root-1"
              >
                <WithStyles(CardContent)>
                  <CardContent
                    classes={
                      Object {
                        "root": "MuiCardContent-root-29",
                      }
                    }
                    component="div"
                  >
                    <div
                      className="MuiCardContent-root-29"
                    >
                      Hi!
                    </div>
                  </CardContent>
                </WithStyles(CardContent)>
              </div>
            </Paper>
          </WithStyles(Paper)>
        </Card>
      </WithStyles(Card)>
    </Home>
  </MyMock>
</Index>
`;

exports[`Index page renders auth Error 1`] = `
<Index
  isAuthenticated={true}
>
  <MyMock>
    <div>
      Some error
    </div>
  </MyMock>
</Index>
`;

exports[`Index page renders auth Loading 1`] = `
<Index
  isAuthenticated={true}
>
  <MyMock>
    <div>
      Loading...
    </div>
  </MyMock>
</Index>
`;

exports[`Index page renders unauth 1`] = `
<Link
  href="/auth/sign-in"
  prefetch={true}
>
  <a>
    Sign In
  </a>
</Link>
`;
